t/errlog.t

```diff
@@ -1025,3 +1025,122 @@ GET /t
 log a warning event
 --- no_error_log
 do not log another warning event
+
+
+
+=== TEST 27: ringbuf overflow bug
+--- http_config
+    lua_capture_error_log 4k;
+--- config
+    location /t {
+        access_by_lua_block {
+            local errlog = require "ngx.errlog"
+            local msg = string.rep("*", 10)
+
+            for i = 1, 10 do
+                ngx.log(ngx.ERR, msg)
+            end
+        }
+
+        content_by_lua_block {
+            local errlog = require "ngx.errlog"
+            local msg = string.rep("*", 10)
+
+            for i = 1, 40 do
+                local res = errlog.get_logs(1)
+                if res and #res then
+                    ngx.log(ngx.ERR, msg)
+                end
+            end
+
+            local res = errlog.get_logs(10)
+            ngx.say("log lines: #", #res / 2)
```

Comment:
The resulting table has the following structure: 
`{ level1, time1, msg1, level2, time2, msg2, ... }`

https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/errlog.md#get_logs

we should use `#res / 3` now.